vMenu/CommonFunctions.cs

```diff
@@ -561,25 +564,27 @@ public static async Task TeleportToCoords(Vector3 pos, bool safeModeDisabled = f
                     // it seem to load the world ANY faster than without, but whatever.
                     RequestCollisionAtCoord(pos.X, pos.Y, z);
 
-                    // Request a new scene. This will trigger the world to be loaded around that area.
-                    NewLoadSceneStart(pos.X, pos.Y, z, pos.X, pos.Y, z, 50f, 0);
+                    // Set focus for that area.
+                    SetFocusPosAndVel(pos.X, pos.Y, pos.Z, 0, 0, 0);
 
-                    // Timer to make sure things don't get out of hand (player having to wait forever to get teleported if something fails).
-                    int tempTimer = GetGameTimer();
+                    // Start to load the scene.
+                    NewLoadSceneStart(pos.X, pos.Y, pos.Z, pos.X, pos.Y, pos.Z, 50f, 0);
 
-                    // Wait for the new scene to be loaded.
-                    while (IsNetworkLoadingScene())
-                    {
-                        // If this takes longer than 1 second, just abort. It's not worth waiting that long.
-                        if (GetGameTimer() - tempTimer > 1000)
-                        {
-                            Log("Waiting for the scene to load is taking too long (more than 1s). Breaking from wait loop.");
-                            break;
-                        }
+                    // Reset the timer.
+                    tempTimer = GetGameTimer();
 
-                        await Delay(0);
+                    // Wait for the scene to be loaded with a timeout.
+                    while (!IsNewLoadSceneLoaded() && Game.GameTime - tempTimer < 10000) // Can take up to 10 seconds
```

Comment:
One more thing I forgot to mention, maybe reduce this back down to say 3 seconds max. Because it's really not worth it for teleports to take that long to load. I'd much rather fall back to other methods of checking Z/loading the surroundings than having to wait 10 seconds for a scene, which I still doubt is actually the best way to force the surroundings to load anyway. I just don't know what a better solution is.

Reply:
Me neither, at your request I will, it was more of a personal choice for my server :p

Reply:
All good :)